src/modules/file-upload-input.marko

```diff
@@ -0,0 +1,122 @@
+<div id="file-upload-input">
+    <section-header metadata=input.metadata name=input.name alpha="true"/>
+
+    <p>
+        The file-upload-input lets the user choose one or more files from their device storage. Once chosen, the files can be uploaded to a server using${" "}
+        <a
+            target="_blank"
+            href="https://developer.mozilla.org/en-US/docs/Learn/Forms"
+        >
+            form submission
+        </a>
+        , or manipulated using JavaScript code and
+        <a
+            target="_blank"
+            href="https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications"
+        >
+            the File API
+        </a>
+        .
+    </p>
+    <p>
+        The entire element can function as a dropzone for drag and drop, however this functionality requires JavaScript and is outside the scope of a CSS library such as eBay Skin.
+    </p>
+
+    <h3>Base</h3>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <div class="file-upload-input">
+                <input
+                    id="file-upload-input-1"
+                    class="file-upload-input__input"
+                    type="file"
+                    multiple
+                >
+                <label
+                    for="file-upload-input-1"
+                    class="file-upload-input__label"
+                >
+                    <div class="file-upload-input__container">
+                        <div class="file-upload-input__upload-icon">
+                            <svg
+                                class="icon file-upload-input__upload-icon-small"
+                                aria-hidden="true"
+                            >
+
+                                <use href="#icon-upload-16"/>
+                            </svg>
+                            <svg
+                                class="icon file-upload-input__upload-icon-large"
+                                aria-hidden="true"
+                            >
+
+                                <use href="#icon-upload-24"/>
+                            </svg>
+                        </div>
+                        <div class="file-upload-input__content">
+                            <span class="file-upload-input__content-header">
+                                Add files
+                            </span>
+                            <span class="file-upload-input__content-subheader">
+                                Up to 10MB per file in JPG, JPEG, PNG, MP4.
```

Comment:
Maybe a little picky, but it doesn't feel like this text should be part of the input's label. It feels more like a description.

I wonder also if a voice dictation user would know what label to speak here? "Add files" or "Browse files"?

Reply:
The more I think about this the more I'm tempted for us to just not do any offscreen trickery with the input and just display the input onscreen right where the "Browse Files" CTA is.

Let's discuss a little more offline.

We should also probably give textual affordance that users can drop files, like lightning does, " or drop files":

<img width="261" alt="Screenshot 2024-09-17 at 9 21 43 AM" src="https://github.com/user-attachments/assets/7cf726a2-8c13-427e-8ee9-5d84a969f877">

Reply:
One advantage of the offscreen trickery and making the input take the entire width and height of the container is being able to take advantage of the drag drop and click to open dialog features of the file input without adding JS to compensate. We can explore the idea of having the input button show and having textual affordance for drop files, would need some input from design